### PR TITLE
[NIC-21] Save watcher record

### PR DESCRIPTION
**Pull request by AksyonovaKaterina**

### CODE REVIEW

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfo.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.nimbusds.jwt.SignedJWT
+import org.springframework.lang.NonNull
+
+/**
+ * @see UserAuthInfoArgumentResolver
+ */
+open class UserAuthInfo(
+
+    @NonNull
+    val authUserId: String,
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfo.kt]: userId

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        try {
+            return SignedJWT.parse(tokenWithoutPrefix)
+        } catch (ex: Exception) {
+            throw InvalidJwtException()
+        }
+    }
+
+    private fun getTokenWithoutBearerPrefix(token: String): String {
+        if (token.startsWith("Bearer ")) {
+            return token.substring("Bearer ".length)
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt]: companion object

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        return token
+    }
+
+    private fun verify(@Suppress("UNUSED_PARAMETER") jwt: SignedJWT) {
+        // do nothing
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt]: redundant method

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@RestController
+@RequestMapping(PUBLIC_API_V1_WATCHER_URL_PATH)
+class WatcherController(
+    private val watcherService: WatcherService,
+) {
+
+    @PostMapping(
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+        produces = [MediaType.APPLICATION_JSON_VALUE],
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt]: static import

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/entity/WatcherEntity.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package com.nosql.watcher.entity
+
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import org.springframework.data.mongodb.core.mapping.Field
+import java.util.Date
+
+@Document(collection = "person")
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/entity/WatcherEntity.kt]: watcher

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/repository/WatcherRepository.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.nosql.watcher.repository
+
+import com.nosql.watcher.entity.WatcherEntity
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.repository.ReactiveMongoRepository
+
+interface WatcherRepository : ReactiveMongoRepository<WatcherEntity, ObjectId>{
+}
```
Comment (V7074) [inline backend/watcher-service/src/main/kotlin/com/nosql/watcher/repository/WatcherRepository.kt:8]: delete curve brackets

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/repository/WatcherRepository.kt (hunk 6 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/repository/WatcherRepository.kt:8]: we will ned them in next task, so it is not critical

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/entity/WatcherEntity.kt (hunk 5 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/entity/WatcherEntity.kt]: fixed

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt (hunk 4 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/controller/WatcherController.kt]: fixed

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt (hunk 3 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt]: fixed

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt (hunk 2 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfoArgumentResolver.kt]: fixed

File: backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfo.kt (hunk 1 above)
Comment (AksyonovaKaterina) [reply backend/watcher-service/src/main/kotlin/com/nosql/watcher/common/auth/UserAuthInfo.kt]: fixed

**Review by V7074 — APPROVED** [review]

**Pull request merged**